sklearn/neural_network/_multilayer_perceptron.py

```diff
@@ -1138,7 +1192,18 @@ def _validate_input(self, X, y, incremental, reset):
         # This downcast to bool is to prevent upcasting when working with
         # float32 data
         y = self._label_binarizer.transform(y).astype(bool)
-        return X, y
+
+        # check sample weight
+        sample_weight = _check_sample_weight(sample_weight, X, dtype=X.dtype)
+        nonzero_indices = np.nonzero(sample_weight)[0]
+        if (
+            nonzero_indices.shape[0] < sample_weight.shape[0]
+        ):  # filter out zero-weighted samples
```

Comment:
I can see this is required to get the common test to pass, but I prefer to get the work correctly with zero weight samples. All the other estimators that have `sample_weights` pass the common tests without actively remove samples.